saleor/graphql/checkout/mutations/checkout_create.py

```diff
@@ -385,6 +424,35 @@ def get_instance(cls, info: ResolveInfo, **data):
             instance.user = user
         return instance
 
+    # Same perform_mutation as base class, but slightly changed
+    # Due to incorrect inheritance model, base method can't be used,
+    # because it's removing metadata from cleaned_input.
+    # However, it's required here.
+    #
+    # TODO: This should be refactored
+    @classmethod
+    def _perform_base_mutation(cls, _root, info: ResolveInfo, /, **data):
+        instance = cls.get_instance(info, **data)
+        data = data.get("input")
+        cleaned_input = cls.clean_input(info, instance, data)
+        metadata_list = cleaned_input.get("metadata", None)
+        private_metadata_list = cleaned_input.get("private_metadata", None)
+        instance = cls.construct_instance(instance, cleaned_input)
```

Comment:
What happens if I pass cleaned_input to construct instance - and it still contains metadata inside? Shouldn't ORM just drop these fields?